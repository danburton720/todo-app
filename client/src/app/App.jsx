import React from 'react';

import TodosContainer from './containers/TodosContainer';
import TasksContainer from './containers/TasksContainer';

export default class App extends React.Component {
    render() {
        return (
            <div>
                <div>
                    <TodosContainer />
                </div>
                <div>
                    <TasksContainer />
                </div>
            </div>
        )
    }
}