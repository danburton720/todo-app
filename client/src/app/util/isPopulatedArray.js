export default (src) => Array.isArray(src) && src.length > 0;