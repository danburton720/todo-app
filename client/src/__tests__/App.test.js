import React from 'react';
import {shallow} from 'enzyme';
import App from '../app/App';
import {socket} from '../app/util/socket';

jest.mock('../app/util/socket', () => ({
    socket: {
        on: jest.fn()
    }
}));

describe('App', () => {
    let shallowWrapper;
    let props;
    const wrapper = () => {
        if (!shallowWrapper) {
            shallowWrapper = shallow(<App {...props}/>);
        }
        return shallowWrapper;
    };

    beforeEach(() => {
        props = {};
        shallowWrapper = undefined;
        jest.clearAllMocks();
    });

    describe('render', () => {
        it('should return ResponsiveNavBar, TodosContainer and TasksContainer', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        describe('mount', () => {
            it('should call socket.on with \'connect\' and a function', () => {
                expect(socket.on).not.toHaveBeenCalled();
                wrapper();
                expect(socket.on).toHaveBeenCalledTimes(1);
                expect(socket.on).toHaveBeenCalledWith('connect', expect.any(Function));
            });
        });
    });
});