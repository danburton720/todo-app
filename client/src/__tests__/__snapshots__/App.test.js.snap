// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render should return ResponsiveNavBar, TodosContainer and TasksContainer 1`] = `
<div
  className="app"
>
  <ResponsiveNavBar />
  <div
    className="sideBar"
    id="sideBar"
  >
    <Connect(TodosContainer) />
  </div>
  <div
    className="main"
  >
    <Connect(TasksContainer) />
  </div>
</div>
`;
